go/app/server.go

```diff
@@ -150,16 +200,40 @@ func (s *Handlers) AddItem(w http.ResponseWriter, r *http.Request) {
 // storeImage stores an image and returns the file path and an error if any.
 // this method calculates the hash sum of the image as a file name to avoid the duplication of a same file
 // and stores it in the image directory.
-func (s *Handlers) storeImage(image []byte) (filePath string, err error) {
-	// STEP 4-4: add an implementation to store an image
-	// TODO:
-	// - calc hash sum
-	// - build image file path
-	// - check if the image already exists
-	// - store image
-	// - return the image file path
+// func (s *Handlers) storeImage(image []byte) (filePath string, err error) {
+// STEP 4-4: add an implementation to store an image
+// TODO:
+// - calc hash sum
+// - build image file path
+// - check if the image already exists
+// - store image
+// - return the image file path
+func (s *Handlers) storeImage(image []byte) (string, error) {
+	// SHA-256ハッシュを計算
+	hasher := sha256.New()
+	_, err := hasher.Write(image)
+	if err != nil {
+		return "", fmt.Errorf("failed to calculate hash: %w", err)
+	}
+	hashSum := hex.EncodeToString(hasher.Sum(nil))
+	fileName := hashSum + ".jpg"
```

Comment:
We are accepting both `.jpeg` and `.jpg` above, but hardcoding to `.jpg` here. That is probably not ideal.

For now, to keep things simple, we should accept only `.jpg` in line 116 above